Userland/Libraries/LibCore/EventLoopImplementationUnix.cpp

```diff
@@ -505,12 +514,34 @@ bool EventLoopManagerUnix::unregister_timer(int timer_id)
 
 void EventLoopManagerUnix::register_notifier(Notifier& notifier)
 {
-    ThreadData::the().notifiers.set(&notifier);
+    auto& thread_data = ThreadData::the();
+
+    thread_data.notifier_by_ptr.set(&notifier, thread_data.poll_fds.size());
+    thread_data.notifier_by_index.append(&notifier);
+    thread_data.poll_fds.append({
+        .fd = notifier.fd(),
+        .events = notification_type_to_poll_events(notifier.type()),
+        .revents = 0,
+    });
```

Comment:
I think i'm on board with why we need two lists here for the notifiers. But, the names are a bit confusing.

Am I reading this correctly that the first list is a map from Notifier --> Index in the pollfds structure, and the second list is a list of notifiers that follows the indexes of the pollfd structure?

Reply:
Yes. This is like that because I can't store anything else in pollfd, obviously. I'm open to naming suggestions though.

Reply:
eh. I suppose it's fine. just takes a look at this function for it to make sense.